### PR TITLE
Bind component animation to new resource model

### PR DESCRIPTION
**Pull request by Erick9Thor**

### CODE REVIEW

File: Source/src/importers/ModelImporter.cpp [hunk 1]
```diff
@@ -25,7 +25,8 @@ void Hachiko::ModelImporter::Import(const char* path, YAML::Node& meta)
     const std::string model_output_path = StringUtils::Concat(model_path.parent_path().string(), "\\", model_path.filename().replace_extension(MODEL_EXTENSION).string());
 
     unsigned flags = aiProcess_CalcTangentSpace | aiProcess_GenSmoothNormals | aiProcess_JoinIdenticalVertices | aiProcess_ImproveCacheLocality | aiProcess_LimitBoneWeights
-                     | aiProcess_SplitLargeMeshes | aiProcess_Triangulate | aiProcess_GenUVCoords | aiProcess_SortByPType | aiProcess_FindDegenerates | aiProcess_FindInvalidData | 0;
+                     | aiProcess_SplitLargeMeshes | aiProcess_Triangulate | aiProcess_GenUVCoords | aiProcess_SortByPType | aiProcess_FindDegenerates | aiProcess_FindInvalidData 
+                     | aiProcess_GlobalScale | 0;
```
Comment (Bernatmago) [inline Source/src/importers/ModelImporter.cpp:29]: making a bitwise or with 0 makes literally no sense 👀

**Review by Bernatmago — APPROVED** [review]

File: Source/src/importers/ModelImporter.cpp (hunk 1 above)
Comment (baransrc) [inline Source/src/importers/ModelImporter.cpp:29]:
```suggestion
                     | aiProcess_GlobalScale;
```

File: Source/src/importers/AnimationImporter.cpp [hunk 2]
```diff
@@ -30,22 +30,126 @@ void Hachiko::AnimationImporter::ImportWithMeta(const char* path, YAML::Node& me
...
+
+    size_bytes = sizeof(unsigned) * second_header.size();
+    memcpy(cursor, &second_header[0], size_bytes);
+    cursor += size_bytes;
+
+    size_bytes = animation->GetName().length();
+    memcpy(cursor, animation->GetName().c_str(), size_bytes);
+    cursor += size_bytes;
+
+    i = 0;
```
Comment (baransrc) [inline Source/src/importers/AnimationImporter.cpp:73]:
```suggestion
```

File: Source/src/importers/AnimationImporter.cpp [hunk 3]
```diff
@@ -30,22 +30,126 @@ void Hachiko::AnimationImporter::ImportWithMeta(const char* path, YAML::Node& me
...
+    {
+        second_header[i] = it->first.length();
+        second_header[i + 1] = it->second.num_positions;
+        second_header[i + 2] = it->second.num_rotations;
+        file_size += it->first.length() + sizeof(float3) * it->second.num_positions + sizeof(Quat) * it->second.num_rotations;
+        i += 3;
+    }
+    file_size += sizeof(unsigned) * second_header.size();
+    
+    const auto file_buffer = new char[file_size];
```
Comment (baransrc) [inline Source/src/importers/AnimationImporter.cpp:57]:
```suggestion
    char* file_buffer = new char[file_size];
```

**Review by baransrc — CHANGES_REQUESTED** [review]

**Review by baransrc — APPROVED** [review]

**Comment by VBoschMatas** [comment]
Engine crashes when I try to import a model that has no animations. I tried to delete the .meta and library but still does the same

**Pull request merged**